### ISSUE TITLE
jax.scipy.special.beta returns nan / wrong sign at multiple defined points (negative integers, ±inf, -0)

### ISSUE THREAD

**Issue by OE-GOD** [jax-ml/jax#38270]
## Description

`jax.scipy.special.beta` returns `nan` (or wrong sign) at multiple inputs where SciPy returns finite or signed-infinity values. The bug surfaces across three distinct edge classes — negative integer arguments, infinite arguments, and the `-0` sign edge.

## Reproducer

```python
import jax
jax.config.update('jax_enable_x64', True)
import jax.scipy.special as jss
import jax.numpy as jnp
import scipy.special as ss
import numpy as np
```

```python
# Edge class 1: wrong sign at -0
print(ss.beta(-0.0, 1.0))                          #  +inf
print(float(jss.beta(jnp.float32(-0.0), jnp.float32(1.0))))  #  -inf  ← BUG (wrong sign)

# Edge class 2: nan at negative integer arguments
for x in [-1.0, -2.0, -3.0]:
    s = float(ss.beta(x, 1.0))
    j = float(jss.beta(jnp.float32(x), jnp.float32(1.0)))
    print(f"beta({x:5}, 1.0): scipy={s}, jax={j}")
# scipy:  -1.0, -0.5, -0.333
# jax:    nan, nan, nan  ← BUG (should match scipy: 1/x)

# Edge class 3: nan at infinite arguments
print(ss.beta(np.inf, 1.0), float(jss.beta(jnp.float32(np.inf), jnp.float32(1.0))))    # 0   vs nan ← BUG
print(ss.beta(-np.inf, 1.0), float(jss.beta(jnp.float32(-np.inf), jnp.float32(1.0))))  # +inf vs nan ← BUG
```

## Expected behaviour

`beta(x, y) = Gamma(x) * Gamma(y) / Gamma(x+y)`. For `y = 1`, this simplifies to `1/x`. SciPy returns the correct values at all the failing inputs above:

| Input            | SciPy `beta(x, 1)`   | JAX           |
|------------------|----------------------|---------------|
| `beta(-0.0, 1.0)` | `+inf`              | `-inf` ← sign error |
| `beta(-1.0, 1.0)` | `-1.0`               | `nan`         |
| `beta(-2.0, 1.0)` | `-0.5`               | `nan`         |
| `beta(-3.0, 1.0)` | `-1/3`               | `nan`         |
| `beta(+inf, 1.0)` | `0.0`                | `nan`         |
| `beta(-inf, 1.0)` | `+inf`               | `nan`         |

Three distinct failure modes, all in the same function, all reproducible by varying one argument. Suggests a single shared edge-case handling path.

## Environment

- jax 0.10.1
- scipy (current)
- `jax_enable_x64 = True`
- CPU, macOS Apple Silicon

## Related

Sixth `jax.scipy.special` mismatch found in the family-of-bugs sweep that produced #38260, #38261, #38262, #38263 (and one I retracted, #38264). This one differs from those four — they were NaN-propagation / pole-value-loss; this is sign-error AND finite-value-loss AND infinity-handling, all at well-defined inputs.

**Comment by beanscg** [comment]
I checked the current `beta` implementation. `jax.scipy.special.beta` promotes arguments, rejects complex values, computes `sign = gammasgn(a) * gammasgn(b) * gammasgn(a + b)`, and returns `sign * lax.exp(betaln(a, b))` at `jax/_src/scipy/special.py:406-432`. `betaln` is the shared base implementation starting at line 308.

The issue's `y=1` cases reduce mathematically to `1/x`, so the failing values are mostly pole/sign/infinity edges around `betaln`/`gammasgn`, not general beta-function numerics.

A regression table can pin the issue examples against SciPy:
- `beta(-0.0, 1.0)` -> `+inf`
- `beta(-1.0, 1.0)` -> `-1.0`
- `beta(-2.0, 1.0)` -> `-0.5`
- `beta(np.inf, 1.0)` -> `0.0`
- `beta(-np.inf, 1.0)` -> `+inf`

**Comment by jakevdp** [comment]
This falls under #38651; please note that it may take a while for this to receive attention from the team.
